curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dfa22d1b521be39a3de7be0.md

```diff
@@ -36,9 +36,20 @@ assert(
 The link's text should be `cat photos`. You have either omitted the text or have a typo.
 
 ```js
-const nestedAnchor = document.querySelector(`p > a`);
+const nestedAnchor = document.querySelector('p > a');
+const innerContent = nestedAnchor.innerHTML;
+assert(
+	/^\s*cat photos\s*$/.test(innerContent)
+);
+```
+
+The link's text should be `cat photos`. You either have leading or ending whitespaces.
```

Comment:
I'm concerned that this hint text has the potential to be confusing. The first thing it says is that the text should be "cat photos" but it is at this point, because we enforced that on the previous test. Yes, `cat photos` as displayed in the hint doesn't have the extra white space, but people aren't realizing that to begin with, which is why we are adding this test. I'm not sure everyone is going to realize that the leading and ending white space referred to in the hint is inside of the anchor. I could see people removing the white space before and after the anchor element instead.

I think we should leave out the first part of this hint and just directly address the extra white space issue, making it clearer that the white space they need to remove is inside the anchor element. I'm not exactly sure how to word that yet, so please feel free to make suggestions.

Reply:
maybe

```suggestion
The text inside your anchor element has extra leading or trailing whitespace. The only space in the anchor text should be between the word `cat` and the word `photos`.
```

Lines 39-52 seem like the only things this PR should be touching.

Reply:
How about these hint ?
------------------------------
1. The linked text should be `cat photos` without any leading or ending whitespace inside the anchor tag.

2. The linked text should be `cat photos` without any spacing before `cat` or after `photos`.

Reply:
That's sorta the opposite of "I think we should leave out the first part of this hint"?

Reply:
> maybe
> 
> ```suggestion
> The text inside your anchor element has extra leading or trailing whitespace. The only space in the anchor text should be between the word `cat` and the word `photos`.
> ```
> 
> Lines 39-52 seem like the only things this PR should be touching.

In which step? 12 or 13 or both the steps?
In my opinion it should be on both the steps because 13th step is about deleting the second anchor tag outside the p tag but what if the user changes something in the anchor tag inside the p tag.

Reply:
> That's sorta the opposite of "I think we should leave out the first part of this hint"?

Ohh I misinterpreted that somehow. I thought he meant to leave that and change the next part.

Reply:
A PR should do 1 thing. You don't need to go touching other files to update Step 12.

Reply:
> A PR should do 1 thing. You don't need to go touching other files to update Step 12.

Well I didn't want to touch other files it was requested by @lasjorg

Reply:
Your first commit started messing around with Step 13 instead of Step 12: https://github.com/freeCodeCamp/freeCodeCamp/pull/51991/commits/76f5334b8e766a1f1ca2be2b82867789457ad341

Reply:
> what if the user changes something in the anchor tag inside the p tag.

You could ask that question about any step really. We can only test for so much. If we wanted to add this white space test to another step, I think it would be step 14, since the user is actually modifying this anchor element by adding a `target` attribute to it, so I suppose there is a greater possibility they could change the text inside the anchor as well.

But since step 13 does not involve changing the anchor in the `p` element then I don't think the white space test is needed for it.

Reply:
> Your first commit started messing around with Step 13 instead of Step 12: https://github.com/freeCodeCamp/freeCodeCamp/pull/51991/commits/76f5334b8e766a1f1ca2be2b82867789457ad341 
> 
> The issue only requested a change about Step 12. No other step.

Oh is that so
So finally I have to make these following changes:

1. Change the hint description for the whitespaces test to this `The text inside your anchor element has extra leading or trailing whitespace. The only space in the anchor text should be between the word `cat` and the word `photos`.`

2. Add the more readable tests written by @lasjorg to the step-12 file only.

Is this it?

Reply:
I would 

1) Undo your changes to Step 13. They are unrelated to the issue.

2) Update the whitespace hint as above

3) Make any readability fixes needed

Reply:
I think clarifying the assert message as suggested by @jeremylt is a good idea.

I agree the tests are much more relevant to step 12 than 13. I don't particularly have an issue with testing for it in step 13 as well, but as said we can't account for everything.

The issue opened was about step 13 and I don't see a problem with this PR changing the tests for both steps.

Reply:
Ah, we didn't update the description of the issue

> They shouldn't be messing with that p element in the first place in step 13. I think step 12 is more relevant for this issue.